src/popover/popover.js

```diff
@@ -361,10 +361,20 @@ class Popover extends React.Component<PopoverPropsT, PopoverPrivateStateT> {
       return null;
     }
 
-    const isValidElement = React.isValidElement(anchor);
-    const anchorProps = this.getAnchorProps();
+    let anchorProps = this.getAnchorProps();
+
+    if (typeof anchor === 'object' && React.isValidElement(anchor)) {
+      if (typeof anchor.type === 'function') {
```

Comment:
Reading undocumented values on the react element is putting this component at risk of regressions on react minor version changes. Silently not forwarding the ref is going to lead to more unexpected behavior

Reply:
Yeah fair enough let me see if I can come up with something

Reply:
👍 Thanks for looking into the problem. I'm happy to land a solution, but hopefully in a more maintainable way

Reply:
Yeah no problems 👍